bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/graphics/ImageFileProvider.java

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+ * Copyright (c) 2018, 2025 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *     Hannes Wellmann - introduce ImageFileProvider as replacement for ImageFileNameProvider
+ *******************************************************************************/
+package org.eclipse.swt.graphics;
+
+import java.nio.file.Path;
+
+/**
+ * Interface to provide a callback mechanism to get information about images
+ * when the application is moved from a low DPI monitor to a high DPI monitor.
+ * This provides API which will be called by SWT during the image rendering.
+ *
+ * This interface needs to be implemented by client code to provide the image
+ * information on demand.
+ *
+ * @since 3.129
+ */
+public interface ImageFileProvider {
```

Comment:
Theoretically we could also just use a `IntFunction<Path>` instead, but this would make especially the documentation inconvenient in many places and we might loose some additional context, so I think a dedicated interface is fine for it.

On the other hand, this interface could at least extend `IntFunction<Path>` and default implement its `apply()` method.
```suggestion
public interface ImageFileProvider implements IntFunction<Path> {
```

Reply:
What would be the benefit of implementing that interface? It is not that expressive, so I like the semantically precise implementation that is proposed more.

Reply:
The advantage is that it could be used directly in existing methods that accept an `IntFunction`, but I agree that we should have a dedicated interface for this purpose.
The remaining question for me is if it should extend `IntFunction`, so that such provider could for example be used directly with an `IntStream`. But I'm not sure if that that's really done that often and worth the side-effects this would have? OTOH, the side-effects are probably not that impactful, I assume?

Reply:
From my point of view, the disadvantage is that it prevents you from declaring a method properly reflecting its semantics (like the current `getImagePath(zoom)`) and gives you a meaningless `apply(value)` instead. Functional interfaces make sense for defining lambas or other anonynous types, but not being implemented by an actual interface or class as usually they are too generic to actually use their methods in a proper OO way.